### PR TITLE
Move location polling options

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

This change is still in discussion so am opening as a draft.

Ref to the change that added LocationPollingOptions https://github.com/mapbox/mapbox-navigation-android/pull/3800

### Changelog

```
<changelog>Move location polling options into NavigationOptions</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 1]
```diff
@@ -163,6 +162,21 @@ class MapboxNavigation(
...
+     * @see [registerTripSessionStateObserver]
+     * @see [registerRouteProgressObserver]
+     */
+    @RequiresPermission(anyOf = [ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION])
+    fun startTripSession() {
+        tripSession.start()
+        notificationChannelField?.let {
+            monitorNotificationActionButton(it.get(null) as ReceiveChannel<NotificationAction>)
+        }
+    }
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:178]: Could this be reverted to where it was before https://github.com/mapbox/mapbox-navigation-android/pull/3800? Why was it moved before `init`? 🤔

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 2]
```diff
@@ -191,8 +205,7 @@ class MapboxNavigation(
         )
         tripSession = NavigationComponentProvider.createTripSession(
             tripService = tripService,
-            locationEngine = navigationOptions.locationEngine,
-            navigatorPredictionMillis = navigationOptions.navigatorPredictionMillis,
+            navigationOptions = navigationOptions,
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:208]: Do we want to pass in the whole `NavigationOptions` object around? I'd rather to pass only what's strictly necessary, even if that means adding one more parameter, I think is worth it, also `NavigationComponentProvider.createTripSession` is `internal`.

**Review by Guardiola31337 — COMMENTED** [review]
Also Metalava is failing

```
/root/code/libnavigation-base/api/current.txt:102: error: Removed class com.******.navigation.base.options.LocationPollingOptions [RemovedClass]
src/main/java/com/******/navigation/base/options/NavigationOptions.kt:41: error: Added method com.******.navigation.base.options.NavigationOptions.getLocationEngineRequest() [AddedMethod]
src/main/java/com/******/navigation/base/options/NavigationOptions.kt:165: error: Added method com.******.navigation.base.options.NavigationOptions.Builder.locationEngineRequest(com.******.android.core.location.LocationEngineRequest) [AddedMethod]
Aborting: Found compatibility problems checking the public API (/root/code/libnavigation-base/src/main/java) against the API in /root/code/libnavigation-base/api/current.txt

> Task :libnavigation-base:checkApi FAILED
```

and some `unit-tests` because of https://github.com/mapbox/mapbox-events-android/issues/506

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 1 above)
Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:178]: 🤔  .. uhh, rebase?

Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: <command><shift><up>

🤦 . must have pulled it up when copying things around

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 2 above)
Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:193]: Honestly wanted to make NavigationOptions available to all of Navigation

**Comment by zugaldia** [comment]
We'll want to backport these changes to a 1.2.x patch release, this week.

/cc: @LukasPaczos

**Comment by Guardiola31337** [comment]
Noting that https://github.com/mapbox/mapbox-navigation-android/pull/3820 landed 🎉 

Rebasing from `master` should solve the tests failures.

cc @kmadsen

**Comment by kmadsen** [comment]
The test that failed is [getStatusImmediatelyAfterUpdateLocation](https://github.com/mapbox/mapbox-navigation-android/blob/3aa80690cca773acfd47a771ffaaa4669a9f54b1/libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionTest.kt#L257-L266)
``` kotlin
    @Test
    fun getStatusImmediatelyAfterUpdateLocation() = coroutineRule.runBlockingTest {
        tripSession.start()

        updateLocationAndJoin()
        val slot = slot<Long>()
        coVerify { navigator.getStatus(capture(slot)) }

        assertTrue("${slot.captured} >= $navigatorPredictionMillis", slot.captured >= navigatorPredictionMillis)
    }
```
Output: `java.lang.AssertionError: 1100 >= 1500`

The test appears to be wrong 🤔 .. because the default `navigatorPredictionMillis` is 1100, not 1500

**Review by LukasPaczos — APPROVED** [review]

**Pull request merged**